packages/main/src/plugin/extension-loader.ts

```diff
@@ -1041,6 +1041,18 @@ export class ExtensionLoader {
       startPod(engineId: string, podId: string): Promise<void> {
         return containerProviderRegistry.startPod(engineId, podId);
       },
+      async statsContainer(
+        engineId: string,
+        containerId: string,
+        callback: (stats: containerDesktopAPI.ContainerStatsInfo) => void,
+      ): Promise<Disposable> {
+        const identifier = await containerProviderRegistry.getContainerStats(engineId, containerId, callback);
+        return Disposable.create(() => {
+          containerProviderRegistry.stopContainerStats(identifier).catch((err: unknown): void => {
+            console.error('Something went wrong while trying to stop container stats', err);
+          });
+        });
+      },
```

Comment:
I think it might be the first call doing that, but I'm wondering if we should not include by default this disposable as part of the extensionContext subscriptions so even if developer forget to add it, it'll be auto-removed on extension deactivation

Reply:
(not required to merge)